lib/base.js

```diff
@@ -34,59 +34,76 @@ Base.prototype.makeRequest = function(options) {
 
     var method = get(options, 'method', 'GET').toUpperCase();
 
+    var url =
+        this._airtable._endpointUrl +
+        '/v' +
+        this._airtable._apiVersionMajor +
+        '/' +
+        this._id +
+        get(options, 'path', '/') +
+        '?' +
+        objectToQueryParamString(get(options, 'qs', {}));
+
+    var controller = new AbortController();
+
     var requestOptions = {
         method: method,
-        url:
-            this._airtable._endpointUrl +
-            '/v' +
-            this._airtable._apiVersionMajor +
-            '/' +
-            this._id +
-            get(options, 'path', '/'),
-        qs: get(options, 'qs', {}),
         headers: this._getRequestHeaders(get(options, 'headers', {})),
-        json: true,
-        timeout: this._airtable.requestTimeout,
+        signal: controller.signal,
     };
+
     if ('body' in options && _canRequestMethodIncludeBody(method)) {
-        requestOptions.body = options.body;
+        requestOptions.body = JSON.stringify(options.body);
```

Comment:
im assuming the `JSON.stringify` was previously handled by the request library?

Reply:
It was.